File: GoldenReports.Application/src/DTOs/Reports/CreateReportDto.cs
namespace GoldenReports.Application.DTOs.Reports;

public record CreateReportDto
{
    public Guid NamespaceId { get; init; }
    
    public string Name { get; init; } = null!;
    
    public string? Description { get; init; }
    
    public string? Query { get; init; }
    
    public string? Styles { get; init; }
    
    public string? Template { get; init; }

    public IEnumerable<ReportParameterDto> Parameters { get; init; } = new List<ReportParameterDto>();

    public IEnumerable<ReportVariableDto> Variables { get; init; } = new List<ReportVariableDto>();
}

File: GoldenReports.UI/old/src/app/core/api/models/create-report-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ReportParameterDto } from './report-parameter-dto';
import { ReportVariableDto } from './report-variable-dto';
export interface CreateReportDto {
  description?: null | string;
  name?: string;
  namespaceId?: string;
  parameters?: Array<ReportParameterDto>;
  query?: null | string;
  styles?: null | string;
  template?: null | string;
  variables?: Array<ReportVariableDto>;
}
